src/npcmove.cpp

```diff
@@ -3653,7 +3653,7 @@ void npc::pick_up_item()
         add_msg_debug( debugmode::DF_NPC, "%s::pick_up_item(); Canceling on player's request", get_name() );
         fetching_item = false;
         wanted_item = nullptr;
-        moves -= 1;
+        mod_moves( -1 );
```

Comment:
not -0.01 * get_speed() ?

Reply:
Honestly it shouldn't even exist, I don't **see** any reason why it needs to modify the NPC's moves at all. But I am assuming there is some weird need to recalculate based on total moves, and that the code is doing something useful.

It should only trigger in the case where the player switches the NPC's rules *while they have the pickup intention* anyway - If the rules are set beforehand, they'll never enter npc::pick_up_item() at all.